pkg/bosh/manifest/manifest.go

```diff
@@ -1,12 +1,202 @@
+type Update struct {
+	Canaries        int     `yaml:"canaries"`
+	MaxInFlight     string  `yaml:"max_in_flight"`
+	CanaryWatchTime string  `yaml:"canary_watch_time"`
+	UpdateWatchTime string  `yaml:"update_watch_time"`
+	Serial          bool    `yaml:"serial,omitempty"`
+	VMStratergy     *string `yaml:"vm_stratergy,omitempty"`
+}
+
+// MigratedFrom from BOSH deployment manifest
+type MigratedFrom struct {
+	Name string `yaml:"name"`
+	Az   string `yaml:"az,omitempty"`
+}
+
+// IPv6 from BOSH deployment manifest
+type IPv6 struct {
+	Enable bool `yaml:"enable"`
+}
+
+// JobDir from BOSH deployment manifest
+type JobDir struct {
+	Tmpfs     *bool  `yaml:"tmpfs,omitempty"`
+	TmpfsSize string `yaml:"tmpfs_size,omitempty"`
+}
+
+// Agent from BOSH deployment manifest
+type Agent struct {
+	Settings string `yaml:"settings,omitempty"`
+	Tmpfs    *bool  `yaml:"tmpfs,omitempty"`
+}
+
+// AgentEnvBoshConfig from BOSH deployment manifest
+type AgentEnvBoshConfig struct {
+	Password              string  `yaml:"password,omitempty"`
+	KeepRootPassword      string  `yaml:"keep_root_password,omitempty"`
+	RemoveDevTools        *bool   `yaml:"remove_dev_tools,omitempty"`
+	RemoveStaticLibraries *bool   `yaml:"remove_static_libraries,omitempty"`
+	SwapSize              *int    `yaml:"swap_size,omitempty"`
+	IPv6                  IPv6    `yaml:"ipv6,omitempty"`
+	JobDir                *JobDir `yaml:"job_dir,omitempty"`
+	Agent                 *Agent  `yaml:"agent,omitempty"`
+}
+
+// AgentEnv from BOSH deployment manifest
+type AgentEnv struct {
+	PersistentDiskFS           string              `yaml:"persistent_disk_fs,omitempty"`
+	PersistentDiskMountOptions []string            `yaml:"persistent_disk_mount_options,omitempty"`
+	AgentEnvBoshConfig         *AgentEnvBoshConfig `yaml:"bosh,omitempty"`
+}
+
 // InstanceGroup from BOSH deployment manifest
 type InstanceGroup struct {
-	Name      string
-	Instances int
+	Name               string                 `yaml:"name"`
+	Instances          int                    `yaml:"instances"`
+	Azs                []string               `yaml:"azs"`
+	Jobs               []Job                  `yaml:"jobs"`
+	VMType             string                 `yaml:"vm_type,omitempty"`
+	VMExtensions       []string               `yaml:"vm_extensions,omitempty"`
+	VMResources        *VMResource            `yaml:"vm_resources"`
+	Stemcell           string                 `yaml:"stemcell"`
+	PersistentDisk     *int                   `yaml:"persistent_disk,omitempty"`
+	PersistentDiskType string                 `yaml:"persistent_disk_type,omitempty"`
+	Networks           []*Network             `yaml:"networks,omitempty"`
+	Update             *Update                `yaml:"update,omitempty"`
+	MigratedFrom       *MigratedFrom          `yaml:"migrated_from,omitempty"`
+	LifeCycle          string                 `yaml:"lifecycle,omitempty"`
+	Properties         map[string]interface{} `yaml:"properties,omitempty"`
```

Comment:
We need interface{}, because the value could be a string, integer or boolean, right?